Makefile.docker

```diff
@@ -3,173 +3,122 @@
+# docker-cilium-image
+$(eval $(call DOCKER_IMAGE_TEMPLATE,docker-cilium-image,Dockerfile,cilium,$(DOCKER_IMAGE_TAG),docker-cilium-manifest))
 
-dev-docker-image-unstripped: NOSTRIP=1
-dev-docker-image-unstripped: UNSTRIPPED=-unstripped
-dev-docker-image-unstripped: dev-docker-image
+# dev-docker-image
+$(eval $(call DOCKER_IMAGE_TEMPLATE,dev-docker-image,Dockerfile,cilium-dev,$(DOCKER_IMAGE_TAG),docker-cilium-dev-manifest))
 
-docker-cilium-dev-manifest:
-	@$(ECHO_CHECK) contrib/scripts/push_manifest.sh cilium-dev $(DOCKER_IMAGE_TAG)
-	$(QUIET) contrib/scripts/push_manifest.sh cilium-dev $(DOCKER_IMAGE_TAG)
+# docker-plugin-image
+$(eval $(call DOCKER_IMAGE_TEMPLATE,docker-plugin-image,cilium-docker-plugin.Dockerfile,docker-plugin,$(DOCKER_IMAGE_TAG),docker-plugin-manifest))
 
-# Build cilium-operator images.
+# docker-hubble-relay-image
+$(eval $(call DOCKER_IMAGE_TEMPLATE,docker-hubble-relay-image,hubble-relay.Dockerfile,hubble-relay,$(DOCKER_IMAGE_TAG),docker-hubble-relay-manifest))
+
+# docker-clustermesh-apiserver-image
+$(eval $(call DOCKER_IMAGE_TEMPLATE,docker-clustermesh-apiserver-image,clustermesh-apiserver.Dockerfile,clustermesh-apiserver,$(DOCKER_IMAGE_TAG),docker-clustermesh-apiserver-manifest))
+
+# docker-opetator-images.
 # We eat the ending of "operator" in to the stem ('%') to allow this pattern
-# to build also 'docker-operator-image', where the stem would be empty otherwise
-docker-opera%-image: GIT_VERSION $(BUILD_DIR)/cilium-opera%.Dockerfile build-context-update
-	$(QUIET)$(CONTAINER_ENGINE) build \
+# to build also 'docker-operator-image', where the stem would be empty otherwise.
+$(eval $(call DOCKER_IMAGE_TEMPLATE,docker-opera%-image,cilium-opera%.Dockerfile,opera%,$(DOCKER_IMAGE_TAG),docker-opera%-manifest))
+
+#
+# This is deprecated and will be removed unless someone protests.
+#
+docker-cilium-image-for-developers:
+	# DOCKER_BUILDKIT allows for faster build as well as the ability to use
+	# a dedicated dockerignore file per Dockerfile.
+	$(QUIET)DOCKER_BUILDKIT=1 $(CONTAINER_ENGINE) build \
 		$(DOCKER_FLAGS) \
-		--build-arg BASE_IMAGE=${BASE_IMAGE} \
-		--build-arg NOSTRIP=${NOSTRIP} \
-		--build-arg LOCKDEBUG=${LOCKDEBUG} \
+		--build-arg LOCKDEBUG=${RACE}\
 		--build-arg RACE=${RACE}\
-		--build-arg CILIUM_SHA=$(firstword $(GIT_VERSION)) \
-		-f $(BUILD_DIR)/cilium-opera$*.Dockerfile \
-		-t cilium/opera$*$(UNSTRIPPED):$(DOCKER_IMAGE_TAG) $(DOCKER_BUILD_DIR)
-	$(QUIET)$(CONTAINER_ENGINE) tag cilium/opera$*$(UNSTRIPPED):$(DOCKER_IMAGE_TAG) cilium/opera$*$(UNSTRIPPED):$(DOCKER_IMAGE_TAG)-${GOARCH}
-	$(QUIET)$(CONTAINER_ENGINE) tag cilium/opera$*$(UNSTRIPPED):$(DOCKER_IMAGE_TAG) $(DOCKER_REGISTRY)/cilium/opera$*$(UNSTRIPPED):$(DOCKER_IMAGE_TAG)
-	@echo "Push like this when ready:"
-	@echo "${CONTAINER_ENGINE} push cilium/opera$*$(UNSTRIPPED):$(DOCKER_IMAGE_TAG)-${GOARCH}"
+		--build-arg V=\
+		--build-arg LIBNETWORK_PLUGIN=\
+		-t $(IMAGE_REPOSITORY)/cilium-dev:latest . -f ./cilium-dev.Dockerfile
```

Comment:
The `--build-arg` arguments look wrong, but also I thought this target was already removed. I say we just drop these lines.

Reply:
Removing `docker-cilium-image-for-developers` target altogether.